File: study/2022/07-W2/wg.shin.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        i, j, k = m + n - 1, m - 1, n - 1
        
        while j >= 0 and k >= 0:
            if nums1[j] < nums2[k]:
                nums1[i] = nums2[k]
                i -= 1
                k -= 1
            else:
                nums1[i] = nums1[j]
                i -= 1
                j -= 1
        while k >= 0:
            nums1[i] = nums2[k]
            i -= 1
            k -= 1
           
  """Runtime: 43 ms, faster than 82.88% of Python3 online submissions for Merge Sorted Array.
Memory Usage: 14 MB, less than 37.51% of Python3 online submissions for Merge Sorted Array."""

File: study/2022/08-W1/wg.shin.dart
class Solution {
  TreeNode? invertTree(TreeNode? root) {
    if(root == null){
        return null;
    }
    root as TreeNode;
    var left = root.left;
    var right = root.right;
    root.left = invertTree(right);
    root.right = invertTree(left);
    return root;
  }
}

/*
Runtime: 395 ms, faster than 100.00% of Dart online submissions for Invert Binary Tree.
Memory Usage: 142.5 MB, less than 100.00% of Dart online submissions for Invert Binary Tree.
*/
